tools/system_libs.py

```diff
@@ -615,6 +615,42 @@ def get_default_variation(cls, **kwargs):
     return super().get_default_variation(eh_mode=eh_mode, **kwargs)
 
 
+class SjLjLibrary(Library):
+  def __init__(self, **kwargs):
+    # Whether we use Wasm EH instructions for SjLj support
+    self.is_wasm = kwargs.pop('is_wasm')
+    super().__init__(**kwargs)
+
+  def get_cflags(self):
+    cflags = super().get_cflags()
+    if self.is_wasm:
+      # DISABLE_EXCEPTION_THROWING=0 is the default, which is for Emscripten
+      # EH/SjLj, so we should reverse it.
+      cflags += ['-s', 'SUPPORT_LONGJMP=wasm',
+                 '-s', 'DISABLE_EXCEPTION_THROWING=1',
+                 '-D__USING_WASM_SJLJ__']
+    else:
+      cflags += ['-s', 'SUPPORT_LONGJMP=emscripten']
+    return cflags
+
+  def get_base_name(self):
+    name = super().get_base_name()
+    # TODO Currently emscripten-based SjLj is the default mode, thus no
+    # suffixes. Change the default to wasm exception later.
+    if self.is_wasm:
+      name += '-wasm'
```

Comment:
libcompiler-rt-wasm.a seems a little ambiguous maybe? 

How about `wasm-exceptions` or `wasm-eh`?

Reply:
I think either is fine; Will wait until we decide on the name `SjLjLibrary` vs. `LongjmpLibrary`, and rename this to match that.